src/NJsonSchema/Infrastructure/TypeExtensions.cs

```diff
@@ -73,9 +73,14 @@ public static string GetDescription(this CachedType type, DescriptionAttributeTy
             else
             {
                 dynamic displayAttribute = attributes.FirstAssignableToTypeNameOrDefault("System.ComponentModel.DataAnnotations.DisplayAttribute");
-                if (displayAttribute != null && !string.IsNullOrEmpty(displayAttribute.Description))
+                if (displayAttribute != null)
                 {
-                    return displayAttribute.Description;
+                    // GetDescription returns null if the Description property on the attribute is not specified.
+                    var description = displayAttribute.GetDescription();
```

Comment:
Is using GetDescription() a breaking change from what we had before?

Reply:
GetDescription() will return the value of the Description field if the ResourceType property on the DisplayAttribute is null, so in this circumstance, the behaviour is the same.

If a ResourceType property is specified on the DisplayAttribute, then GetDescription() will retrieve the value from the ResourceType instead which is the intended behaviour of this change e.g.

```csharp
public class MyClass {
  // No ResourceType specified: DisplayAttribute.GetDescription() returns "Some_Property_Desc"
  [Display(Description = "Some_Property_Desc")]
  public string SomeProperty { get; set; }
}


public class MyOtherClass {
  // ResourceType specified: DisplayAttribute.GetDescription() returns MyResources.Some_Property_Desc
  [Display(ResourceType = typeof(MyResources), Name = "Some_Property_Desc")]
  public string SomeProperty { get; set; }
}
```

Reply:
Committed a check to ensure that the `Description` property is not null before calling GetDescription()